sdk/storage/Azure.Storage.Blobs/tests/BlobBaseClientTests.cs

```diff
@@ -5110,6 +5110,84 @@ public async Task SetMetadataAsync_Error()
                 e => Assert.AreEqual("BlobNotFound", e.ErrorCode));
         }
 
+        [RecordedTest]
+        public async Task SetMetadataAsync_Sort()
+        {
+            await using DisposingContainer test = await GetTestContainerAsync();
+            // Arrange
+            BlobBaseClient blob = await GetNewBlobClient(test.Container);
+            IDictionary<string, string> metadata = new Dictionary<string, string>() {
+                { "a0", "a" },
+                { "a1", "a" },
+                { "a2", "a" },
+                { "a3", "a" },
+                { "a4", "a" },
+                { "a5", "a" },
+                { "a6", "a" },
+                { "a7", "a" },
+                { "a8", "a" },
+                { "a9", "a" },
+                { "_", "a" },
+                { "_a", "a" },
+                { "a_", "a" },
+                { "__", "a" },
+                { "_a_", "a" },
+                { "b", "a" },
+                { "c", "a" },
+                { "y", "a" },
+                { "z", "z_" },
+                { "_z", "a" },
+                { "_F", "a" },
+                { "F", "a" },
+                { "F_", "a" },
+                { "_F_", "a" },
+                { "__F", "a" },
+                { "__a", "a" },
+                { "a__", "a" }
+             };
+
+            // Act
+            Response<BlobInfo> response = await blob.SetMetadataAsync(metadata);
+
+            // Assert
+
+            // Ensure that we grab the whole ETag value from the service without removing the quotes
+            Assert.AreEqual(response.Value.ETag.ToString(), $"\"{response.GetRawResponse().Headers.ETag}\"");
+
+            // Ensure the value has been correctly set by doing a GetProperties call
+            Response<BlobProperties> getPropertiesResponse = await blob.GetPropertiesAsync();
+            AssertDictionaryEquality(metadata, getPropertiesResponse.Value.Metadata);
+        }
+
+        [RecordedTest]
+        public async Task SetMetadataAsync_Sort_InvalidMetadata()
```

Comment:
@jalauzon-msft, @vincenttran-msft, @weirongw23-msft, please take a look at this PR, I think there is a small bug in my implementation of the sorting algorithm.  For this test, I am getting the wrong string to sign.  In the image below, left is what the algorithm is generating, and right is what the service is expecting:

![image](https://github.com/user-attachments/assets/a424b48a-5cdc-4f40-b581-e889ffadbdb0)